From de6332498822840f4873191f773143f83a607a6a Mon Sep 17 00:00:00 2001
From: Julian Kung <julianrkung@berkeley.edu>
Date: Sun, 3 Jan 2021 21:20:37 -0800
Subject: [PATCH] Renamed financialSummary function names

---
 src/lib/airtable/request.js | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/lib/airtable/request.js b/src/lib/airtable/request.js
index a3bcbf7..a05146a 100644
--- a/src/lib/airtable/request.js
+++ b/src/lib/airtable/request.js
@@ -158,7 +158,7 @@ export const createPayment = async (payment, customer) => {
   }
 }
 
-export const createFinancialSummarie = async (record) => {
+export const createFinancialSummary = async (record) => {
   return createRecord(Tables.FinancialSummaries, record);
 };
 
@@ -303,7 +303,7 @@ export const getAllPayments = async (filterByFormula = '', sort = []) => {
   return getAllRecords(Tables.Payments, filterByFormula, sort);
 };
 
-export const getFinancialSummarieById = async (id) => {
+export const getFinancialSummaryById = async (id) => {
   return getRecordById(Tables.FinancialSummaries, id);
 };
 
-- 
2.15.1

