diff --git a/src/lib/airtable/schema.js b/src/lib/airtable/schema.js
index a88bb47..b1a1584 100644
--- a/src/lib/airtable/schema.js
+++ b/src/lib/airtable/schema.js
@@ -27,7 +27,7 @@ export const Columns = {
 		password: {name:`Password`, type:`text`},
 		name: {name:`Name`, type:`text`},
 		paymentIds: {name:`Payments`, type:`foreignKey-many`},
-		meterReadingsAndInvoiceIds: {name:`Meter Readings and Invoices`, type:`foreignKey-many`},
+		meterReadingAndInvoiceIds: {name:`Meter Readings and Invoices`, type:`foreignKey-many`},
 		customerUpdateIds: {name:`Customer Updates`, type:`foreignKey-many`},
 		id: {name:`ID`, type:`formula`},
 		purchaseRequestIds: {name:`Purchase Requests`, type:`foreignKey-many`},
@@ -35,13 +35,20 @@ export const Columns = {
 		inventoryUpdateIds: {name:`Inventory Updates`, type:`foreignKey-many`},
 	},
 	"Sites": {
+		customers: {name:`CustomerData`, type:`custom-object`},
+		financialSummaries: {name: `FinancialSummaries`, type:`custom-object`},
+		tariffPlans: {name:`TariffPlans`, type:`custom-object`},
 		name: {name:`Name`, type:`text`},
 		userIds: {name:`Users`, type:`foreignKey-many`},
 		customerIds: {name:`Customers`, type:`foreignKey-many`},
-		financialSummarieIds: {name:`Financial Summaries`, type:`foreignKey-many`},
+		financialSummaryIds: {name:`Financial Summaries`, type:`foreignKey-many`},
 		tariffPlanIds: {name:`Tariff Plans`, type:`foreignKey-many`},
 		id: {name:`ID`, type:`formula`},
 		inventoryIds: {name:`Inventory`, type:`foreignKey-many`},
+		products: {name: `Products`, type: `custom-object`},
+		inventory: {name: `SiteInventory`, type: `custom-object`},
+		purchaseRequests: {name: `PurchaseRequests`, type: `custom-object`},
+		inventoryUpdates: {name: `InventoryUpdates`, type: `custom-object`},
 	},
 	"Tariff Plans": {
 		name: {name:`Name`, type:`text`},
